src/settings.h

```diff
@@ -217,6 +218,35 @@ class Settings
         ms_registryPath = path;
     }
 
+    /// Set custom configuration read, write and delete functions
+    static void SetConfigMethods(void *customConfigData, 
+                                win_sparkle_config_read_t customConfigRead,
+                                win_sparkle_config_write_t customConfigWrite, 
+                                win_sparkle_config_delete_t customConfigDelete)
+    {
+        CriticalSectionLocker lock(ms_csVars);
+        
+        ms_currentConfigRead = DefaultConfigRead;
+        ms_currentConfigWrite = DefaultConfigWrite;
+        ms_currentConfigDelete = DefaultConfigDelete;
+
+        ms_customConfigData = customConfigData;
+
+        if (customConfigWrite)
```

Comment:
This is dangerous. It allows overriding only some of the functions, but not others — I think this should be all-or-nothing: if you customize config functions, you need to customize all of them.

(An alternative API to drive the point home would be to use not 3 separate arguments, but a "virtual table-like" struct with function pointers.)